backend/canisters/user_index/impl/src/updates/c2c_notify_chit.rs

```diff
@@ -23,7 +23,10 @@ fn c2c_notify_chit_impl(args: Args, state: &mut RuntimeState) -> Response {
         now,
     ) {
         if let Some(user) = state.data.users.get_by_user_id(&user_id) {
-            state.data.chit_leaderboard.update_position(user_id, user.total_chit_earned());
+            state
+                .data
+                .chit_leaderboard
+                .update_position(user_id, user.total_chit_earned(), args.chit_balance);
```

Comment:
This should be based on the `args.timestamp` rather than just taking the current month, otherwise if someone earns some CHIT right at the end of the month that'll be assigned to the new month incorrectly.